heliopy/data/mms.py

```diff
@@ -79,22 +81,136 @@ def available_files(probe, instrument, starttime, endtime, data_rate=''):
     _validate_instrument(instrument)
     probe = _validate_probe(probe)
     _validate_data_rate(data_rate)
+    
     start_date = starttime.strftime('%Y-%m-%d')
-    end_date = endtime.strftime('%Y-%m-%d')
+    if starttime.date() == endtime.date():
+        end_date = (endtime.date() + timedelta(days=1)).strftime('%Y-%m-%d')
+    else:
+        end_date = endtime.strftime('%Y-%m-%d')
 
     query = {}
     query['sc_id'] = 'mms' + probe
     query['instrument_id'] = instrument
     if len(data_rate):
         query['data_rate_mode'] = data_rate
+    if len(product_string):
+        query['descriptor'] = product_string
     query['start_date'] = start_date
     query['end_date'] = end_date
 
     r = requests.get(query_url, params=query)
     files = r.text.split(',')
+    files = filter_time(files, starttime, endtime)
     return files
 
 
+def filter_time(fnames, starttime, endtime):
+    """
```

Comment:
This docstring should have the same format as the ones elsewhere, which is the numpydoc format. https://numpydoc.readthedocs.io/en/latest/ has more info about the standard.